### PR TITLE
feat: Add mainnet staging deployment testing (SC-766)

### PR DESCRIPTION
**Pull request by lucas-manuel**
Note that `DeployEthereumTest` tests will fail in CI because they have to run against a local deployment until the mainnet staging deployment is live. Taken out of CI for this reason with `--no-match-contract` (nmc) flag used.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,12 +70,50 @@ This is a linear rate limit that increases over time with a maximum limit. This
...
+cast rpc --rpc-url="$MAINNET_RPC_URL" anvil_impersonateAccount 0xD1668fB5F690C59Ab4B0CAbAd0f8C1617895052B
+cast send 0x6b175474e89094c44da98b954eedeac495271d0f 'transfer(address,uint256)' $ETH_FROM 100000000000000000000000 --unlocked --from 0xD1668fB5F690C59Ab4B0CAbAd0f8C1617895052B
+```
+
+4. Run the following command to deploy the staging environment against both local forks:
+```
+make deploy-ethereum-staging
+```
+
+5. Copy the files `mainnet-latest.json` and `base-latest.json` from the `script/output/1/` directory to `mainnet-20241017.json` and `base-20241017.json` in the `script/output/1` directory respectively.
```
Comment (hexonaut) [inline README.md:113]: Should be `mainnet-release-<date>.json` and same for base.

File: script/output/1/base-release-20241017.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (hexonaut) [inline script/output/1/base-release-20241017.json:1]: This is not actually released on mainnet and shouldn't be included in the PR. The actual deployment should be added.

File: script/output/1/mainnet-release-20241017.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (hexonaut) [inline script/output/1/mainnet-release-20241017.json:1]: Same here.

File: script/staging/StagingDeploymentBase.sol [hunk 4]
```diff
@@ -178,7 +180,18 @@ contract StagingDeploymentBase is Script {
...
+        // This contract is necessary to get past the `kiss` requirement from the pause proxy.
+        // It wraps the `noFee` calls with regular PSM swap calls.
+        psm = address(new PSMWrapper(USDC, dai, livePsm));
+
+        // NOTE: This is a HACK to make sure that `fill` doesn't get called until the call reverts.
+        //       Because this PSM contract is a wrapper over the real PSM, the controller queries
+        //       the DAI balance of the PSM to check if it should fill or not. Filling with DAI
+        //       fills the live PSM NOT the wrapper, so the while loop will continue until the
+        //       function reverts. Dealing DAI into the wrapper will prevent fill from being called.
+        IERC20(dai).transfer(psm, USDS_UNIT_SIZE * 10);
```
Comment (hexonaut) [inline script/staging/StagingDeploymentBase.sol:194]: This seems like a large amount to burn each time. If unit size is $10 then this will burn $100 every time. Does it need to be this big?

**Review by hexonaut — CHANGES_REQUESTED** [review]
Mostly looks good, but should do an actual deploy on mainnet and add the real release along with updating the test to point at that and removing CI to ignore the mainnet staging test.

File: README.md (hunk 1 above)
Comment (lucas-manuel) [reply README.md]: Using `latest` for now and will update after deployment

File: script/output/1/base-release-20241017.json (hunk 2 above)
Comment (lucas-manuel) [reply script/output/1/base-release-20241017.json]: Removed

File: script/output/1/mainnet-release-20241017.json (hunk 3 above)
Comment (lucas-manuel) [reply script/output/1/mainnet-release-20241017.json]: Removed

File: script/staging/StagingDeploymentBase.sol (hunk 4 above)
Comment (lucas-manuel) [reply script/staging/StagingDeploymentBase.sol]: You're right, setting it equal to unit size is sufficient to pass the check. Updated.

**Review by hexonaut — APPROVED** [review]

**Review by hexonaut — APPROVED** [review]
